### DISCUSSION TITLE
Does StreamPack support - flip/change camera functionality?

### DISCUSSION THREAD

**Discussion by dhruvdroid** [ThibaultBee/StreamPack#149] [General]
Hi Team,

I was going through the implementation of the StreamPack and it's classes. Checked the `CameraRtmpLiveStreamer` & `CameraSrtLiveStreamer` classes as well. 

Didn't find public function to be used for the functionality of changing the camera while live streaming? If anyone has used something similar please do share the resources and articles to refer.

Thanks in advance.

**Comment by dhruvdroid** [comment]
Found the solution of the same via - 
```
<io.github.thibaultbee.streampack.views.PreviewView
            android:id="@+id/preview_view"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            app:enableZoomOnPinch="true"
            app:cameraFacingDirection="back"
            app:layout_constraintBottom_toBottomOf="parent"
            app:layout_constraintEnd_toEndOf="parent"
            app:layout_constraintStart_toStartOf="parent"
            app:layout_constraintTop_toTopOf="parent" />
```

`private val cameraFacingDirection: FacingDirection` provides the option to select the front or back camera by providing the string.

**Reply by ThibaultBee** [reply]
On the streamer, you can also call `streamer.camera = "cameraId"`
